server/events/repo_whitelist_checker.go

```diff
@@ -72,7 +72,21 @@ func (r *RepoWhitelistChecker) matchesRule(rule string, candidate string) bool {
 		return false
 	}
 
-	// Finally we can use the wildcard. Substring both so they're comparing before the wildcard. Example:
+	// If wildcard is not the last character, substring both to compare what is after the wildcard.  Example:
+	// candidate: repo-abc
+	// rule: *-abc
+	// substr(candidate): -abc
+	// substr(rule): -abc
+	if wildcardIdx != len(rule)-1 {
+		// If the rule substring after wildcard does not exist in the candidate, then it is not a match.
+		if !strings.Contains(candidate, rule[wildcardIdx+1:]) {
+			return false
+		}
+
+		return candidate[strings.Index(candidate, rule[wildcardIdx+1:]):] == rule[wildcardIdx+1:]
```

Comment:
I think we need to use `strings.LastIndex` here? Otherwise this test fails:
```go
		{
			"should match with duplicate",
			"*runatlantis",
			"runatlantis/runatlantis",
			"github.com",
			true,
		},
```

Reply:
is this a test you just added ad-hoc?  also not understanding this test rationale (as per my other comment to include github.com at the beginning).  So the reply to that one may also answer this one.  thanks!

Reply:
Yes it's just one I added.